### DISCUSSION TITLE
Identify token transfer in transaction

### DISCUSSION THREAD

**Discussion by 0x-j** [aptos-labs/aptos-developer-discussions#119] [Questions] [answered]
Asked by `sly_of_zero` on discord
> how to identify if a transaction was a token transfer? I need to know the token being transferred, the amount of the token, and the receiver

**Comment by 0x-j** [comment]
I assume you are talking about transferring coin (ERC-20 on Aptos), you can find these info in the event of the tx. Here's an [example](https://explorer.aptoslabs.com/txn/997088103/events?network=testnet) where I transfer APT to another account. It had one `0x1::coin::WithdrawEvent` that withdrew APT from my wallet, and one `0x1::coin::DepositEvent` that deposited to the recipient.

**Reply by slyofzero** [reply]
So my purpose of keeping track of this is to make something like a buybot which tracks any token buys. Would these events mark any token transfer from one address to another? In this case I should be listening for any `0x1::coin::DepositEvent`?

**Reply by gregnazario** [reply, marked as answer]
`0x1::coin::DepositEvent` will tell you when someone deposits tokens to their account.

`0x1::coin::WithdrawEvent` will tell you when someone removes tokens from their account.

In the case of buys or transfers, you will always have a deposit on the user who receives the token, and a withdrawal from the other user.  It is possible if the coins don't go through the `CoinStore` that no withdraw event is emitted

Here's a more detailed doc on how to track coin events https://aptos.dev/guides/system-integrators-guide/#tracking-coin-balance-changes
